cipher/rsa/rsa_test.go

```diff
@@ -35,30 +35,27 @@ var rsaTestData = []struct {
 	},
 }
 
-func TestEncryptDecrypt(t *testing.T) {
+func testPrecondition(t *testing.T) (int64, int64, int64) {
 	// Both prime numbers
 	p := int64(61)
 	q := int64(53)
```

Comment:
```suggestion
func testPrecondition(t *testing.T) (int64, int64, int64) {
	t.Helper()
	// Both prime numbers
	p := int64(61)
	q := int64(53)
```
mark this as a helper function. That way you won't have to add "Algorithm failed in preamble stage" everytime an error happens.